File: 1950-sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.ts
function arraySign(nums: number[]): number {
    let negativeCount = 0;
    for (const num of nums) {
        if (num === 0) {
            return 0;
        } else if (num < 0) {
            negativeCount++;
        }
    }
    return negativeCount % 2 === 0 ? 1 : -1;
}

File: 1950-sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.rs
impl Solution {
    pub fn array_sign(nums: Vec<i32>) -> i32 {
        nums.iter().fold(1, |acc, &x| {
            if x == 0 {
                return 0;
            } else if x < 0 {
                -acc
            } else {
                acc
            }
        })
    }
}
